### ISSUE TITLE
Add a number of examples for the use of conditions in MDF

### ISSUE THREAD

**Issue by pgleeson** [ModECI/MDF#275]
There is one core example illustrating some simple condition handling in MDF: https://github.com/ModECI/MDF/blob/conditions_example/examples/MDF/README.md#conditions, but since there are multiple types of conditions in graph scheduler would be good to add more example showing the use of these in MDF graphs. 

I've added a  folder for these here: https://github.com/ModECI/MDF/tree/conditions_example/examples/MDF/conditions. @Ivy8127 will take the lead in adding these here (I've pointed her at an example added in [a recent PR](https://github.com/ModECI/MDF/pull/266/files#diff-7cab305a6a78540131fcce1fafb912ccf6b52aa50b4a0a2b794aea71004d5a95R180)), but if you have any more @kmantel, please feel free to add these here.

**Comment by kmantel** [comment]
@Ivy8127 For reference, there's a list of all current conditions in the graph-scheduler documentation at https://kmantel.github.io/graph-scheduler/Condition.html#list-of-pre-specified-conditions, and there are a lot of examples in tests at https://github.com/kmantel/graph-scheduler/blob/main/tests/scheduling/test_condition.py

I think Threshold, EveryNCalls, and TimeInterval seem more commonly useful or good for examples.

@pgleeson if you have certain patterns/types of execution that you or others are interested in, I could help with translating those into Conditions

**Comment by Ivy8127** [comment]
> @Ivy8127 For reference, there's a list of all current conditions in the graph-scheduler documentation at https://kmantel.github.io/graph-scheduler/Condition.html#list-of-pre-specified-conditions, and there are a lot of examples in tests at https://github.com/kmantel/graph-scheduler/blob/main/tests/scheduling/test_condition.py
> 
> I think Threshold, EveryNCalls, and TimeInterval seem more commonly useful or good for examples.
> 
> @pgleeson if you have certain patterns/types of execution that you or others are interested in, I could help with translating those into Conditions

Version 1.1.0 of graph-scheduler has updated threshold conditions. Trying to run my threshold example locally and it throws the Threshold condition type not supported exception. Do I manually change the setup.cfg to version 1.1.0? It currently has this version ```graph_scheduler>=1.0.0 ```  @pgleeson

**Comment by pgleeson** [comment]
Yes, I had to do an update from graph_scheduler 1.0.0 locally to get it to work. Please do change the minima dependency to version 1.1.0 in setup.cfg.

**Comment by Ivy8127** [comment]
Current error from the execution engine on line 1233 https://github.com/ModECI/MDF/blob/main/src/modeci_mdf/execution_engine.py#:~:text=else%3A-,raise%20ValueError(,),-return%20args, not sure where I'm going wrong. This is when I try to run my threshold condition example
```raise ValueError(
ValueError: No param_A evaluable object for Node(metadata=None, id='A', input_ports=[InputPort(metadata=None, id='input_port1', shape=None, type=None)], 
functions=[],
 parameters=[Parameter(metadata=None, id='param_A', value='param_A + 1', default_initial_value=0, time_derivative=None, function=None, args=None, conditions=[])], 
output_ports=[OutputPort(metadata=None, id='output_1', value='param_A', shape=None, type=None)]), 
options: ['input_port1', 'param_A', 'output_1']```

**Comment by kmantel** [comment]
> > @Ivy8127 For reference, there's a list of all current conditions in the graph-scheduler documentation at https://kmantel.github.io/graph-scheduler/Condition.html#list-of-pre-specified-conditions, and there are a lot of examples in tests at https://github.com/kmantel/graph-scheduler/blob/main/tests/scheduling/test_condition.py
> > I think Threshold, EveryNCalls, and TimeInterval seem more commonly useful or good for examples.
> > @pgleeson if you have certain patterns/types of execution that you or others are interested in, I could help with translating those into Conditions
> 
> Version 1.1.0 of graph-scheduler has updated threshold conditions. Trying to run my threshold example locally and it throws the Threshold condition type not supported exception. Do I manually change the setup.cfg to version 1.1.0? It currently has this version `graph_scheduler>=1.0.0 ` @pgleeson

Just to install it locally, you can upgrade with `pip install --upgrade graph-scheduler`, it's just that MDF doesn't strictly need the threshold condition for its own tests. 



> Current error from the execution engine on line 1233 https://github.com/ModECI/MDF/blob/main/src/modeci_mdf/execution_engine.py#:~:text=else%3A-,raise%20ValueError(,),-return%20args, not sure where I'm going wrong. This is when I try to run my threshold condition example
> 
> ```
> ValueError: No param_A evaluable object for Node(metadata=None, id='A', input_ports=[InputPort(metadata=None, id='input_port1', shape=None, type=None)], 
> functions=[],
>  parameters=[Parameter(metadata=None, id='param_A', value='param_A + 1', default_initial_value=0, time_derivative=None, function=None, args=None, conditions=[])], 
> output_ports=[OutputPort(metadata=None, id='output_1', value='param_A', shape=None, type=None)]), 
> options: ['input_port1', 'param_A', 'output_1']```
> ```

This problem should have been fixed in #280

**Comment by pgleeson** [comment]
Done
